core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -187,20 +204,29 @@ public TaskExecutor(
             int guaranteedNumberOfDriversPerTask,
             int maximumNumberOfDriversPerTask,
             VersionEmbedder versionEmbedder,
+            Duration interruptRunawaySplitsTimeout,
+            Predicate<List<StackTraceElement>> interruptibleSplitPredicate,
+            Duration interruptSplitInterval,
             MultilevelSplitQueue splitQueue,
             Ticker ticker)
     {
         checkArgument(runnerThreads > 0, "runnerThreads must be at least 1");
         checkArgument(guaranteedNumberOfDriversPerTask > 0, "guaranteedNumberOfDriversPerTask must be at least 1");
         checkArgument(maximumNumberOfDriversPerTask > 0, "maximumNumberOfDriversPerTask must be at least 1");
         checkArgument(guaranteedNumberOfDriversPerTask <= maximumNumberOfDriversPerTask, "guaranteedNumberOfDriversPerTask cannot be greater than maximumNumberOfDriversPerTask");
+        checkArgument(interruptRunawaySplitsTimeout.getValue(TimeUnit.SECONDS) >= 1.0, "interruptRunawaySplitsTimeout must be at least 1 second");
 
         // we manage thread pool size directly, so create an unlimited pool
         this.executor = newCachedThreadPool(threadsNamed("task-processor-%s"));
         this.executorMBean = new ThreadPoolExecutorMBean((ThreadPoolExecutor) executor);
         this.runnerThreads = runnerThreads;
+
         this.versionEmbedder = requireNonNull(versionEmbedder, "versionEmbedder is null");
 
+        this.interruptRunawaySplitsTimeout = interruptRunawaySplitsTimeout;
+        this.interruptibleSplitPredicate = interruptibleSplitPredicate;
+        this.interruptSplitInterval = interruptSplitInterval;
```

Comment:
Rename fields to match args. Make all fields not-null. Use `Optional` when applicable.

Reply:
I'd just put all this in one object which can be made optional